CubeFS Project Security Self-Assessment - Security Pals

avinashnarasimhan18 opened the pull request:
Created and added the first draft of the CubeFS Project Security Self-Assessment. Please feel free to share your feedback on the security self-assessment.

ragashreeshekar commented on assessments/projects/cubefs/self-assessment.md:
```diff
@@ -0,0 +1,300 @@
+| Website | https://cubefs.io/ |
+| Security Provider | No  |
+| Languages | Golang |
+| SBOM | [CubeFS SBOM File](https://github.com/avinashnarasimhan18/tag-security/blob/4e6b73248f7807cef59a456ab87d8f357094f380/assessments/projects/cubefs/cubefs_sbom.json)|
```
This link seems to be from your personal fork of the repository.
Please update it with the link from the project.

avinashnarasimhan18 replied:
The SBOM Section has been updated.

avinashnarasimhan18 commented:
@eddie-knight Thank you for starting the review on the self-assessment and giving us some initial comments.

CubeFS has not made the SBOM publicly available, so our team generated one using the FOSSA CLI. Would you prefer we keep the SBOM or remove it and mention that it is not publicly available? Please let us know.

avinashnarasimhan18 commented:
@ragashreeshekar We have started working on your review comments. Please let us know if there are any other changes to be made. We will change accordingly and push all of them in one commit.

ragashreeshekar commented on assessments/projects/cubefs/self-assessment.md:
```diff
@@ -0,0 +1,300 @@
+  * Actor Interaction: Volume Management System manages access permissions and interacts with container/client applications to grant or restrict access to volumes. Regular audits to ensure access permissions align with security policies. 
+
+
+### Goals
```
Goals section usually talks about what the project aims to accomplish (could incl. security goals but that isn't the main focus). A good example is https://github.com/cncf/tag-security/blob/main/assessments/projects/kyverno/self-assessment.md#goal

avinashnarasimhan18 replied:
The goals section has been updated as per the review comment.

ragashreeshekar commented on assessments/projects/cubefs/self-assessment.md:
```diff
@@ -0,0 +1,300 @@
+  * **Goal**: Safeguard shared keys and support regular key rotation for enhanced security.
+  * **Security Guarantee**: Future enhancements to support key rotation, reducing the risk of key compromise and strengthening encryption mechanisms.
+
+### Non-goals
```
Non-goals section usually talks about what actions are out of scope for the same. A good example is https://github.com/cncf/tag-security/blob/main/assessments/projects/kyverno/self-assessment.md#non-goals

avinashnarasimhan18 replied:
The non-goals section has been updated as per the review comment.

ragashreeshekar commented on assessments/projects/cubefs/self-assessment.md:
```diff
@@ -0,0 +1,300 @@
+* **Volume Access Control** -
+  * Actors: Volume Management System, Container/Client Applications
+  * Actor Interaction: Volume Management System manages access permissions and interacts with container/client applications to grant or restrict access to volumes. Regular audits to ensure access permissions align with security policies. 
+
```
What are the components involved and workflow in the Data read operations?

avinashnarasimhan18 replied:
The data read operation has been expounded upon.

ragashreeshekar commented on assessments/projects/cubefs/self-assessment.md:
Considering the project is in cloud storage domain, pointers on data backup, recovery could be useful to include here.

ragashreeshekar commented on assessments/projects/cubefs/self-assessment.md:
Are there any SAST tools for container image vulnerabilities, static code analysis or source composition analysis included in their CI pipeline? Please include here.

avinashnarasimhan18 replied:
Lack of SAST tools in CubeFS CI/CD pipeline has been addressed.

ragashreeshekar commented:
> @ragashreeshekar We have started working on your review comments. Please let us know if there are any other changes to be made. We will change accordingly and push all of them in one commit.

Thanks for the PR @avinashnarasimhan18 and team, appreciate the efforts. 
I have completed first pass of review. Please feel free to reach out here or on slack for any questions and clarifications. 
Along with addressing the comments, kindly update the PR branch with the latest content in the repo as this branch is out-of-date with the base branch.

avinashnarasimhan18 commented:
> > @ragashreeshekar We have started working on your review comments. Please let us know if there are any other changes to be made. We will change accordingly and push all of them in one commit.
> 
> Thanks for the PR @avinashnarasimhan18 and team, appreciate the efforts. I have completed first pass of review. Please feel free to reach out here or on slack for any questions and clarifications. Along with addressing the comments, kindly update the PR branch with the latest content in the repo as this branch is out-of-date with the base branch.

Thank you @ragashreeshekar. We will start working on your comments right away and update once it's done. We will ensure the branch is updated as well.

avinashnarasimhan18 commented:
@ragashreeshekar , me and my team have addressed your and Eddie's review comments. Please take a look and let us know if there are any other changes you would like us to make.

eddie-knight commented on assessments/projects/cubefs/self-assessment.md:
```diff
@@ -0,0 +1,257 @@
+| Website | https://cubefs.io/ |
+| Security Provider | No  |
+| Languages | Golang |
+| SBOM | Currently, no SBOM is publicly available for CubeFS|
```
Recommended minor change: highlight that target state is not yet achieved, while removing red-herring questions about a potential privately available SBOM.
 
```suggestion
| SBOM | CubeFS does not currently generate an SBOM on release |
```

avinashnarasimhan18 replied:
@eddie-knight, we have added the change you suggested. Please let us know if it's okay.

eddie-knight commented on assessments/projects/cubefs/self-assessment.md:
```suggestion
```

Following discussions from the review team, we are asking that Security Pals please remove any recommendations from the scope of this document. You may optionally communicate recommendations for projects in a more accessible location, such as in a project community channel or GitHub Issue.

avinashnarasimhan18 replied:
@eddie-knight, per your suggestion, we have removed the action items section from this document. We will figure out how to communicate our suggestions to the CubeFS team.

avinashnarasimhan18 commented on assessments/projects/cubefs/self-assessment.md:
```diff
@@ -0,0 +1,300 @@
+Express Learning course provided by Linux Foundation Training & Certification: 
+[Security Assessments for Open Source Projects](https://training.linuxfoundation.org/express-learning/security-self-assessments-for-open-source-projects-lfel1005/).
+
+# Self-assessment outline
```
Template text has been removed and additional heading has been removed.

ragashreeshekar commented:
Thank you for the updates. Looks good to me. 
Tagging @eddie-knight for your review, and @JustinCappos both for your opinion on including maintainers for their ack and review of this work, followed by chair review (as applicable) and merging them to the repo.

JustinCappos commented on assessments/projects/cubefs/self-assessment.md:
```diff
@@ -0,0 +1,235 @@
+# CubeFS Self-assessment
+This Self-assessment document has been created by students at NYU Tandon to help provide information regarding CubeFS, determining gaps in their security, and preparing any security
```
Do we want to change this lead in?  I would think that it should mention the Security Pals effort.

avinashnarasimhan18 replied:
@JustinCappos do you have any suggestions for this?

JustinCappos replied:
Please use: 

This assessment was created by community members as part of the [Security Pals](https://github.com/cncf/tag-security/issues/1102) process, and is currently pending changes from the maintainer team.

avinashnarasimhan18 replied:
Thank you @JustinCappos! The suggested change has been implemented. Please review.

ragashreeshekar commented on assessments/projects/cubefs/self-assessment.md:
```diff
@@ -0,0 +1,234 @@
+# CubeFS Self-assessment
+This Self-assessment document was created by community members as part of the [Security Pals](https://github.com/cncf/tag-security/issues/1102) process, and is currently pending changes from the maintainer team. It aims to help provide information regarding CubeFS, determining gaps in their security, and preparing any security documentation for their users. 
```
```suggestion
This Self-assessment document was created by community members as part of the [Security Pals](https://github.com/cncf/tag-security/issues/1102) process, and is currently pending changes from the maintainer team. 

It aims to help provide information regarding CubeFS, determining gaps in their security, and preparing any security documentation for their users. 

```

ragashreeshekar commented on assessments/projects/cubefs/self-assessment.md:
```diff
@@ -0,0 +1,237 @@
+## Metadata
+
+|   |  |
+| -- | -- |
```
```suggestion
| -- | -- |
| Assessment Stage | Incomplete | 
```

torinvdb commented on assessments/projects/cubefs/self-assessment.md:
```diff
@@ -0,0 +1,238 @@
+# CubeFS Self-assessment
+This Self-assessment document was created by community members as part of the [Security Pals](https://github.com/cncf/tag-security/issues/1102) process, and is currently pending changes from the maintainer team. 
+
+It aims to help provide information regarding CubeFS, determining gaps in their security, and preparing any security documentation for their users. 
```
```suggestion
This Self-assessment document was created by community members as part of the [Security Pals](https://github.com/cncf/tag-security/issues/1102) process and is currently pending changes from the maintainer team. 

The intention of this assessment is to help provide security-related information, such as gap identification and user security considerations, that pertain to the CubeFS project.
```
@JustinCappos
